packages/p2p/src/components/my-ads/my-ads-row-renderer.jsx

```diff
@@ -269,19 +292,37 @@ const MyAdsRowRenderer = observer(({ row: advert, setAdvert }) => {
                                     className='p2p-my-ads__table-popovers__activate'
                                     message={localize('Activate')}
                                 >
-                                    <Icon icon='IcUnarchive' color={general_store.is_barred && 'disabled'} size={16} />
+                                    <Icon icon='IcUnarchive' color={general_store.is_barred && 'disabled'} />
                                 </Popover>
                             </div>
                         )}
-                        <div onClick={onClickEdit}>
-                            <Popover
-                                alignment='bottom'
-                                className='p2p-my-ads__table-popovers__edit'
-                                message={localize('Edit')}
+                        {is_advert_active ? (
+                            <div
+                                onClick={
+                                    enable_action_point && floating_rate_store.rate_type !== rate_type
+                                        ? onClickSwitchAd
+                                        : onClickEdit
+                                }
                             >
-                                <Icon icon='IcEdit' size={16} />
-                            </Popover>
-                        </div>
+                                <Popover
+                                    alignment='bottom'
+                                    className='p2p-my-ads__table-popovers__deactivate'
+                                    message={localize('Edit')}
+                                >
+                                    <Icon icon='IcEdit' />
+                                </Popover>
+                            </div>
+                        ) : (
+                            <div onClick={onClickEdit}>
+                                <Popover
+                                    alignment='bottom'
+                                    className='p2p-my-ads__table-popovers__edit'
+                                    message={localize('Edit')}
+                                >
+                                    <Icon icon='IcEdit' />
+                                </Popover>
+                            </div>
+                        )}
```

Comment:
The Popover component is written twice and instead could be refactored into a single component since they both have the same props, only difference is the `onClick` function prop that is passed, could the code be refactored into something like this instead?
```
const onClickPopoverHandler = () => {
     if (is_advert_active) {
          if (enable_action_point && floating_rate_store.rate_type !== rate_type) {
               return onClickSwitchAd
          }
     }
     return onClickEdit
}
<div onClick={onClickPopoverHandler()}>
    <Popover
         alignment='bottom'
         className='p2p-my-ads__table-popovers__deactivate'
         message={localize('Edit')}
    >
         <Icon icon='IcEdit' />
      </Popover>
</div>
```

Reply:
I will refactor it